nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -4,20 +4,41 @@
 import java.util.Arrays;
 
 import generic.continues.GenericFactory;
+import ghidra.app.util.bin.BinaryReader;
 import ghidra.app.util.bin.ByteProvider;
 import ghidra.app.util.bin.format.FactoryBundledWithBinaryReader;
 import ghidra.app.util.bin.format.pe.PortableExecutable.SectionLayout;
+import ghidra.program.model.data.DataType;
+import nsis.format.NsisScriptHeader;
 
+/**
+ * 
+ * This class represents a Nsis Executable.
+ *
+ */
 public class NsisExecutable {
 
 	public static final String NAME = "NULLSOFT_SCRIPTABLE_INSTALLER_SYSTEM";
 
 	private FactoryBundledWithBinaryReader reader;
-	private long header_offset;
+	private NsisScriptHeader scriptHeader;
+	private long headerOffset;
 
+	/**
+	 * Use createNsisExecutable to create a Nsis Executable object
+	 */
 	public NsisExecutable() {
 	}
 
+	/**
+	 * Creates and initializes a Nsis Executable object
+	 * 
+	 * @param factory      that will be used to create Nsis Executable
+	 * @param byteProvider object that will permit reading bytes from the file
```

Comment:
Could you please document the expectations on lifespan of byteProvider? Can the caller of NsisExecutable call .close() on the provider and continue using NsisExecutable? Who is responsible for calling ByteProvider.close(), caller or NsisExecutable?

Reply:
From what I understand, the lifespan is controlled by Ghidra itself. The byteprovider object is giver to our plugin by Ghidra and Ghidra controlls closing files

Reply:
Added info